hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -529,7 +559,7 @@ public Response head(
     }
 
     ResponseBuilder response = Response.ok().status(HttpStatus.SC_OK)
-        .header("ETag", "" + key.getModificationTime())
+        .header(ETAG, "" + wrapInQuotes(key.getMetadata().get(ETAG)))
```

Comment:
We should also remove the header to avoid it showing up in the customer headers later down the code or filter out the headers that should not be part of custom headers.